internal/services/voiceservices/voice_services_communications_gateway_resource.go

```diff
@@ -0,0 +1,618 @@
+				string(communicationsgateways.E911TypeDirectToEsrp),
+				string(communicationsgateways.E911TypeStandard),
+			}, false),
+		},
+
+		"platforms": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(communicationsgateways.CommunicationsPlatformOperatorConnect),
+					string(communicationsgateways.CommunicationsPlatformTeamsPhoneMobile),
+				}, false),
+			},
+		},
+
+		"service_locations": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"primary_region_properties": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"operator_addresses": {
+									Type:     pluginsdk.TypeList,
+									Required: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+									},
+								},
+								"allowed_media_source_address_prefixes": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+									},
+								},
+
+								"allowed_signaling_source_address_prefixes": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+									},
+								},
+
+								"esrp_addresses": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"auto_generated_domain_name_label_scope": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  string(communicationsgateways.AutoGeneratedDomainNameLabelScopeTenantReuse),
+			ValidateFunc: validation.StringInSlice([]string{
+				string(communicationsgateways.AutoGeneratedDomainNameLabelScopeTenantReuse),
+				string(communicationsgateways.AutoGeneratedDomainNameLabelScopeSubscriptionReuse),
+				string(communicationsgateways.AutoGeneratedDomainNameLabelScopeResourceGroupReuse),
+				string(communicationsgateways.AutoGeneratedDomainNameLabelScopeNoReuse),
+			}, false),
+		},
+
+		"api_bridge": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsJSON,
+		},
+
+		"emergency_dial_strings": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"on_prem_mcp_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+
+		"tags": commonschema.Tags(),
+
+		"teams_voicemail_pilot_number": {
```

Comment:
The `microsoft_teams_voicemail_pilot_number` is a string since the [type of API](https://github.com/Azure/azure-rest-api-specs/blob/56435032b69cc7ebb5c1052b66c7865d0c48f33c/specification/voiceservices/resource-manager/Microsoft.VoiceServices/stable/2023-01-31/voiceservices.json#L966) is string.